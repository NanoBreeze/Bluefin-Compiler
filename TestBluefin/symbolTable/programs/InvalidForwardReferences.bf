struct Another {
	void f() {
		a = 1;
	}
	int a;
};

int main() {
	a = 3;
	int a;
	
	{
		b = 1;
		f();
	}
	First first;
}
void f() {}
float b;
struct First{};