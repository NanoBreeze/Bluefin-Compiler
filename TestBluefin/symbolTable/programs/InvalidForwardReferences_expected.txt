declare - Another - c_struct - t_Another
setCurrentScope - Level 1
resolve - void - c_builtinType - t_void
declare - f - c_function - t_void
enterScope - Level 2
exitScope - Level 2
resolve - int - c_builtinType - t_int
declare - a - c_var - t_int
exitScope - Level 1
resolve - int - c_builtinType - t_int
declare - main - c_function - t_int
enterScope - Level 1
resolve - int - c_builtinType - t_int
declare - a - c_var - t_int
enterScope - Level 2
exitScope - Level 2
resolve - First - UNRESOLVED
exitScope - Level 1
resolve - void - c_builtinType - t_void
declare - f - c_function - t_void
enterScope - Level 1
exitScope - Level 1
resolve - float - c_builtinType - t_float
declare - b - c_var - t_float
declare - First - c_struct - t_First
setCurrentScope - Level 1
exitScope - Level 1
resolve - a - c_var - t_int
resolve - a - c_var - t_int
resolve - a - ILLEGAL_FORWARD_REFERENCE
resolve - b - c_var - t_float
resolve - b - ILLEGAL_FORWARD_REFERENCE
resolve - f - c_function - t_void
resolve - f - ILLEGAL_FORWARD_REFERENCE
